### PR TITLE
[ISSUE #1429]Update Broker's Topic Read or Write Perm failed.

### PR DESCRIPTION
**Pull request by zongtanghu**
## What is the purpose of the change

This pr will resolve the issue that updating broker's topic reading or writing perm fails.

## Brief changelog
The resolved of codes is in this pr.

## Verifying this change
I add the correct logical codes of udpating broker reading or writing perms.

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by zongtanghu** [comment]
Hi @duhengforever,@ShannonDing, please help to review this pr.

**Comment by jonnxu** [comment]
LGTM

**Comment by RongtongJin** [comment]
LGTM~

File: tools/src/main/java/org/apache/rocketmq/tools/command/topic/UpdateTopicPermSubCommand.java [hunk 1]
```diff
@@ -67,46 +71,91 @@ public Options buildCommandlineOptions(Options options) {
...
             topicConfig.setPerm(perm);
             if (commandLine.hasOption('b')) {
-                String addr = commandLine.getOptionValue('b').trim();
-                defaultMQAdminExt.createAndUpdateTopicConfig(addr, topicConfig);
-                System.out.printf("update topic perm from %s to %s in %s success.%n", oldPerm, perm, addr);
-                System.out.printf("%s%n", topicConfig);
-                return;
+                String brokerAddr = commandLine.getOptionValue('b').trim();
+                List<BrokerData> brokerDatas = topicRouteData.getBrokerDatas();
+                boolean isBrokerLegal = false;
```
Comment (duhenglucky) [inline tools/src/main/java/org/apache/rocketmq/tools/command/topic/UpdateTopicPermSubCommand.java]: Does it really need the isBrokerLegal variable here?

Comment (zongtanghu) [reply tools/src/main/java/org/apache/rocketmq/tools/command/topic/UpdateTopicPermSubCommand.java]: @duhenglucky  Hi duheng,I think it's necessary to need the isBrokerLegal variable!

**Pull request merged**